packages/daemon/daemon-docs.md

```diff
@@ -0,0 +1,112 @@
+- this includes establishing an ipc channel which Endo applications can use to establish a connection
+  `makeEndoClient` to create a new connection.
+- The powers accessible within this new context are:
+
+  - `crypto`
+  - `net`
+  - `fs`
+  - `path`
+  - `popen`
+  - `url`
+
+- takes one argument, `locator`. If nothing is passed in, this defaults to [`defaultLocator`](#defaultlocator).
+
+### `terminate`
+
+- creates a new endo client using [makeEndoClient](#makeendoclient) for the sole purpose of gaining access to the `terminate` method.
+- Uses `terminate` in order to sever any existing connections.
+- takes one argument, `locator`. If nothing is passed in, this defaults to [`defaultLocator`](#defaultlocator).
+
+```js
+export const terminate = async (locator = defaultLocator) => {
+  const { resolve: cancel, promise: cancelled } = makePromiseKit();
+  const { getBootstrap, closed } = await makeEndoClient(
+    'harbinger',
+    locator.sockPath,
+    cancelled,
+  );
+  const bootstrap = getBootstrap();
+  await E(bootstrap)
+    .terminate()
+    .catch(() => {});
+  // @ts-expect-error zero-argument promise resolve
+  cancel();
+  await closed.catch(() => {});
+};
+```
+
+### `clean`
+
+- recursively and forcibly removes any directories that have been created throughout the lifecycle of the previous Endo daemon.
+- takes one argument, `locator`. If nothing is passed in, this defaults to [`defaultLocator`](#defaultlocator).
+
+### `stop`
+
+- terminates existing Endo daemon processess before calling [`clean`](#clean).
+- takes one argument, `locator`. If nothing is passed in, this defaults to [`defaultLocator`](#defaultlocator).
+
+### `makeEndoClient`
+
+- Used to spin up a new application within the context of an Endo daemon.
+- The processess created with makeEndoClient come pre-assembled with the ability to communicate over CapTP using Node.js-based messages.
+- Utilizes the global `net` package to create a socket connection for an Endo process. (`net.connect(sockPath)`)
+
+### defaultLocator
+
+- default `locator` object which an Endo process will be initialized with.
+- See [@endo/daemon](https://github.com/endojs/endo/blob/master/packages/daemon/index.js#L26)
+
+```js
+const defaultLocator = {
+  statePath: whereEndoState(process.platform, process.env, info),
+  ephemeralStatePath: whereEndoEphemeralState(
+    process.platform,
+    process.env,
+    info,
+  ),
+  sockPath: whereEndoSock(process.platform, process.env, info),
+  cachePath: whereEndoCache(process.platform, process.env, info),
+};
+```
+
+# Questions
+
+- the `main` function seems is where “all of it” starts.
+- where does `main0` come from?
+- Regarding the locator...
+  - what is the difference between:
+    - statePath
+    - ephemeralStatPath
+    - sockPath
+    - cachePath
+  - after coming across the term "noncelocator" a number of times while reading erights.org, I'm wondering what is the locator within the context of SwingSet?
```

Comment:
I don’t know about these things but there might be a convergent reason for them. My impression is that these are different locators. The Endo pet daemon locator is just for allowing various components to rendezvous and communicate with each other.